### ISSUE TITLE
idea

### ISSUE THREAD

**Issue by atherdon** [ChickenKyiv/recipe-search-react#63]
# Description/Steps to reproduce
@chauhannishith i'm working on npm module that will store our static data for search options.
Can you help me with it?

# Expected result

# Additional information

**Comment by chauhannishith** [comment]
Can you elaborate more about this?

**Comment by atherdon** [comment]
sure, this is why i started this discussion - to help me move it forward.
ok, so for now we have a data folder inside our project. Yes, later we'll
not need it, because we'll connect to api. but a lot of my projects have a
similar import data that we're using for testing purposes. and i think by
movin it into a separated package - is a good thing.
repo is here: https://github.com/GroceriStar/static-data


On 19-Jun-2018 at 10:13 AM, chauhannishith wrote:


Can you elaborate more about this?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ChickenKyiv/recipe-search-react/issues/63#issuecomment-398348536>,
or mute the thread

**Comment by chauhannishith** [comment]
I am trying to understand this but it is taking time

**Comment by chauhannishith** [comment]
Just to be sure, you are talking about creating packages, right?

**Comment by atherdon** [comment]
yes. npm package that can be installed via npm and clean up code of our projects

**Comment by chauhannishith** [comment]
can you tell me what is the status of this and what this does right now and also the final expected result

**Comment by atherdon** [comment]
Ok, sure.

So, as you maybe noticed - we have a lot of repositories right now.
and a lot of this repositories have `data` folders. In this folders we have arrays, that we use as a test cases. It can be named a mocking data. 
So, recipe search form has arrays, that used for displaying options inside selects.
search server using arrays for importing test data for displaying seach results
groceristar repo contain ultimate grocery list that we importing in our database.

A lot of this js arrays are pretty similar. And main idea was to remove this arrays from main repo - this will clean up the code - and import data from npm module.

Second part - we have an import script to our loopback servers. and we have a first version that i wrote 1 year ago at ChickenKyiv server. second version - groceristar server, third and fourth version at recipe-search-api. So i want to test in one place an import script and don't copy-paste a similar code between repositories.

tell me if this make it clear for you.
I can explain more, with details - if you wish

**Comment by chauhannishith** [comment]
So you want to create functions in packages that take the link of api and serve the requested data

**Comment by chauhannishith** [comment]
something like this if i understood correctly

var Data = require('Groceristar')
Data.setDataBase(databaselink)
var allergies = getData(allergies)
push(collection, dataObj) function to add data in database

**Comment by atherdon** [comment]
nope. i want to create a functions that will return data from static files

**Comment by atherdon** [comment]
for now i want https://github.com/ChickenKyiv/recipe-search-react/blob/master/src/components/SearchForm/Ingredient/Ingredient.js#L5
but from separated npm package. so if i'll update the data inside it - we'll be able to upgrade all repositories and have a newest versions

**Comment by chauhannishith** [comment]
ok from files but the functionalities which are said are correct?

**Comment by atherdon** [comment]
yeah, just without a database link for now.
i have a `new` idea - maybe it'll require to update whole servers logic :)
at my PC groceristar main repo is strangely crashed and i decide - maybe this is a good sign to move away from REST to graphQL approach

**Comment by atherdon** [comment]
main issue - why i stuck and decide to ask for help - i didn't figure out how to debug that npm package.
so it'll be cool if you help - because i think this is not a complex task - but it'll give a huge benefit

**Comment by atherdon** [comment]
So, in order to make it more simple: for a first time - i'll be cool to have installed that package at this repository, data folder removed from this repo and everything working as before.
I add some information here: https://github.com/GroceriStar/static-data/issues/14

**Comment by chauhannishith** [comment]
https://github.com/chauhannishith/groceristar-fetch/tree/master

here's the package that you told me to create.
Have a look at it and if you feel like there need to be some changes then tell me.

**Comment by atherdon** [comment]
@chauhannishith  ok i like it, did you test it at this repo?
should i publish it on npm?

**Comment by chauhannishith** [comment]
I tried fetching the data which works and a sample demo separately. Haven't used it to get the data in this repo but that will work too.

You can publish it but you might have to make some changes in package.json

**Comment by atherdon** [comment]
what do you think about transfering this repo to groceristar organization?
for sure i can commit to your repo - but i'll be easy for me to keep all in
one place.


On 02-Jul-2018 at 09:43 AM, chauhannishith wrote:


I tried fetching the data which works and a sample demo separately. Haven't
used it to get the data in this repo but that will work too.

You can publish it but you might have to make some changes in package.json

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ChickenKyiv/recipe-search-react/issues/63#issuecomment-401737843>,
or mute the thread

**Comment by chauhannishith** [comment]
yes i made this separately so that i can directly transfer it. Where should i transfer it?

**Comment by atherdon** [comment]
on form transfer - move to username "GroceriStar"


On 02-Jul-2018 at 09:49 AM, chauhannishith wrote:


yes i made this separately so that i can directly transfer it. Where should
i transfer it?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ChickenKyiv/recipe-search-react/issues/63#issuecomment-401740579>,
or mute the thread

**Comment by atherdon** [comment]
ok, will grab a cup of tea and publish it on npm, so we'll be able to test
how it works on a real case


On 02-Jul-2018 at 09:49 AM, chauhannishith wrote:


yes i made this separately so that i can directly transfer it. Where should
i transfer it?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ChickenKyiv/recipe-search-react/issues/63#issuecomment-401740579>,
or mute the thread

**Comment by chauhannishith** [comment]
after you publish it tell me what name you gave it so that I can add it in this repo

**Comment by atherdon** [comment]
we'll keep this name, for luck :)


On 02-Jul-2018 at 10:01 AM, chauhannishith wrote:


after you publish it tell me what name you gave it so that I can add it in
this repo

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ChickenKyiv/recipe-search-react/issues/63#issuecomment-401744839>,
or mute the thread

**Comment by atherdon** [comment]
- [ ] Install package
- [ ] connect to search form elements
- [ ] delete `data` folder

**Comment by atherdon** [comment]
GroceriStar/static-data#14

**Comment by chauhannishith** [comment]
> @groceristar/groceristar-fetch (1.0.0) was published

are you sure this is the name? It is still not showing

**Comment by chauhannishith** [comment]
is @ also included in the name?

**Comment by chauhannishith** [comment]
I found it but I think webpack has some issue with fs, will try to find a solution to it or replace it in the package

**Comment by atherdon** [comment]
yeah, there new rules with aliases: `yarn add @groceristar/groceristar-fetch` should work

**Comment by atherdon** [comment]
maybe we can use some default options for reading json? i mean in Angular
you just fetch data from it by passing an url, not using file streams as we
do...


On 03-Jul-2018 at 05:45 AM, chauhannishith wrote:


I found it but I think webpack has some issue with fs, will try to find a
solution to it or replace it in the package

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ChickenKyiv/recipe-search-react/issues/63#issuecomment-402019893>,
or mute the thread

**Comment by atherdon** [comment]
@chauhannishith try this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse

But still I like your idea with `fs` - because later we'll need to import csv file into our database - so this approach will be useful

**Comment by atherdon** [comment]
i think you pushed an updates to npm

**Comment by chauhannishith** [comment]
Yeah, I commented the fs part so that you could use it in future if required and replaced it with simple JSON. also, there was an error in production build due to ES6 syntax so changed it to ES5

**Comment by atherdon** [comment]
can you move that error to groceristar-fetch - so we can address it later?


On 03-Jul-2018 at 11:15 AM, chauhannishith wrote:


Yeah, I commented the fs part so that you could use it in future if
required and replaced it with simple JSON. also, there was an error in
production build due to ES6 syntax so changed it to ES5

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ChickenKyiv/recipe-search-react/issues/63#issuecomment-402115713>,
or mute the thread

**Comment by chauhannishith** [comment]
I corrected it in version 1.1.1 so there are not much chances of it arising any time soon at least for a few months

**Comment by atherdon** [comment]
got it

**Comment by atherdon** [comment]
got it. tell me when it'll work fine at search work - excited to check it


On 03-Jul-2018 at 11:19 AM, chauhannishith wrote:


I corrected it in version 1.1.1 so there are not much chances of it arising
any time soon at least for a few months

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ChickenKyiv/recipe-search-react/issues/63#issuecomment-402117260>,
or mute the thread

**Comment by chauhannishith** [comment]
I added this in the form and have removed the data folder

**Comment by atherdon** [comment]
ok, will review it and close this task, btw - don't forget about adding
some tests to groceristar-fetch package


On 03-Jul-2018 at 12:03 PM, chauhannishith wrote:


I added this in the form and have removed the data folder

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ChickenKyiv/recipe-search-react/issues/63#issuecomment-402131310>,
or mute the thread
